Documentation/ApiOverview/TypoScriptSyntax/TypoScriptParserApi/CustomConditions.rst

```diff
@@ -7,90 +7,171 @@
 Implementing Custom Conditions
 ==============================
 
-Now we know how to parse TypoScript and the only thing we still want
-to do is to implement support for custom conditions. As stated in a few
-places *the evaluation* of a condition is external to TypoScript and
-all you need to do in order to have an external process deal with
-conditions is to pass an object as the second parameter to the parse-
-function. This is done in the code listing below:
-
-.. code-block:: php
-
-    class myConditions {
-        /**
-        * Evaluates, if the condition line was "[TYPO3 IS GREAT]".
-        *
-        * @param string $conditionLine The condition line
-        * @return boolean value
-        */
-        public function match($conditionLine) {
-            if ($conditionLine === '[TYPO3 IS GREAT]') {
-                return true;
-            }
-        }
-    }
-    $matchObj = GeneralUtility::makeInstance(\myConditions::class);
-
-    $TSparserObject = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\TYPO3\CMS\Core\TypoScript\Parser\TypoScriptParser::class);
-    $TSparserObject->parse($tsString, $matchObj);
-
-    debug($TSparserObject->setup);
-
-In this example, we define a very simple class with a :php:`match()` function
-inside. This :php:`match()` function must exist and take a string as its
-argument. It is expected to return a boolean value. This
-function should be programmed to evaluate the condition line according
-to your specifications. Currently, if a condition line contains the
-value "[TYPO3 IS GREAT]" then the condition will evaluate to true and
-the subsequent TypoScript will be parsed.
-
-To enable this condition, we simply create an instance of our class
-and pass it to the TypoScript parser upon callings the :php:`parse()`
-method.
-
-Let's test the custom condition class from the code listing
-above. This is done by parsing this TypoScript code:
+In order to provide custom conditions, its essential to understand how
+conditions are written. Refer to :ref:`typoscript-syntax-conditions-syntax` if
+syntax of conditions is not known yet.
 
-.. code-block:: typoscript
+Conditions are evaluated by the `Symfony Expression Language`_ and are evaluated
+to boolean results. Therefore an integrator can write :ts:`[true === true]`
+which would evaluate to true. In order to provide further functionality within
+conditions, the Symfony Expression Language needs to be extended. There are two
+parts that can be added to the language, which are variables and functions.
 
-   someOtherTS = 123
+The following section explain how to add variables and functions.
 
-   [TYPO3 IS GREAT]
+.. _registering-new-provider-within-extension:
 
-   message = Yes
-   someOtherTS = 987
+Registering new Provider within Extension
```

Comment:
this whole section doesn't really belong in this manual as it's nothing to do with typoscript or the parser but is the same no matter where expression language is used (the provider registration etc...) - as expression language is part of ext:form, site configuration and typoscript (and in time probably more places) we should add this chapter to TYPO3 explained instead. The first paragraph here might then simply link to that chapter.

Reply:
I've created a new section where only TypoScript conditions are added for now.